widget-src/components/log/LinkForm.tsx

```diff
@@ -0,0 +1,266 @@
+        return "A link label (40 character maximum) and valid url (including 'https://') are required.";
+      case (labelError):
+        return "A link label (40 character maximum) is required.";
+      case (urlError):
+        return "A valid url (including 'https://') is required.";
+      default:
+        return "";
+    }
+  }
+
+  return (
+    <AutoLayout
+      width="fill-parent"
+      direction="vertical"
+      spacing={GAP.sm}
+      horizontalAlignItems="end"
+      verticalAlignItems="center"
+    >
+      <AutoLayout
+        width="fill-parent"
+        horizontalAlignItems="end"
+        verticalAlignItems="center"
+        spacing={GAP.md}
+      >
+        <Input 
+          name="LinkLabel"
+          fill={COLOR.black}
+          inputFrameProps={{
+            fill: COLOR.white,
+            stroke: !!changeLog.state?.linkFormError?.label ? COLOR.red : COLOR.grey,
+            strokeWidth: SPACE.one,
+            cornerRadius: RADIUS.xs,
+            padding: { horizontal: PADDING.xs, vertical: PADDING.xs }
+          }}
+          placeholder="Link Label"
+          width={SPACE.lg}
+          lineHeight={FONT.lineHeight.xs}
+          fontSize={FONT.size.xs}
+          fontFamily={FONT.family}
+          value={changeLog.state?.link?.label || ''}
+          onTextEditEnd={(e) => {
+            const trimmedLabel = e.characters.trim()
+            updateChange({
+              state: {
+                ...changeLog.state,
+                link: {
+                  ...changeLog.state?.link,
+                  label: trimmedLabel,
+                  url: changeLog.state?.link?.url || '',
+                  key: changeLog.state?.link?.key || '',
+                },
+                linkFormError: {
+                  label: !validLabel(trimmedLabel),
+                  url: !!changeLog.state?.linkFormError?.url,
+                }
+              },
+            })
+          }}
+        />
+        <Input 
+          name="LinkUrl"
+          fill={COLOR.black}
+          inputFrameProps={{
+            fill: COLOR.white,
+            stroke: !!changeLog.state?.linkFormError?.url ? COLOR.red : COLOR.grey,
+            strokeWidth: SPACE.one,
+            cornerRadius: RADIUS.xs,
+            padding: { horizontal: PADDING.xs, vertical: PADDING.xs }
+          }}
+          width="fill-parent"
+          placeholder="https://figma.com"
+          lineHeight={FONT.lineHeight.xs}
+          fontSize={FONT.size.xs}
+          fontFamily={FONT.family}
+          value={changeLog.state?.link?.url || ''}
+          onTextEditEnd={(e) => {
+            const trimmedUrl = e.characters.trim();
+            updateChange({
+              state: {
+                ...changeLog.state,
+                link: {
+                  ...changeLog.state?.link,
+                  label: changeLog.state?.link?.label || '',
+                  url: trimmedUrl,
+                  icon: assignIcon(trimmedUrl),
```

Comment:
@ryansrofe wanted to clarify something with you about how this currently works - at this point I'm actually passing the svg string into the link data - not a reference string like "jira" or "figma". That means that changing the icon within the widget won't actually change the icon associated with the link as that only happens during link creation.

Pros to this approach - if we choose to unsupport/remove an icon in the future it won't break existing links using that icon. 
Cons to this approach - we can't change the icon after it has been associated with the link meaning that if we wanted to update the icon styling in the future we would risk breaking the layout of existing links.

It wouldn't be that hard to refactor this so it uses reference strings instead of full svg strings if we feel that is a more flexible approach going forward. Thoughts?

Reply:
Per our conversation moved this to a string assignment based on url patterns and am adding the svg strings when the link is displayed. That will give us the ability to add/remove icons in the future if needed.